crates/provider/src/native.rs

```diff
@@ -0,0 +1,271 @@
+use std::{
+    self,
+    error::Error,
+    fmt::Debug,
+    process::{Command, Output},
+};
+
+use async_trait::async_trait;
+use serde::Serialize;
+
+use crate::shared::{
+    constants::{DEFAULT_DATA_DIR, DEFAULT_REMOTE_DIR, LOCALHOST},
+    provider::Provider,
+    types::{NamespaceDef, NamespaceMetadata, RunCommandOptions, RunCommandResponse},
+};
+
+pub trait FileSystem {
+    fn create_dir(&mut self, path: impl Into<String>) -> Result<(), Box<dyn Error>>;
+    fn write(
+        &mut self,
+        path: impl Into<String>,
+        content: impl Into<String>,
+    ) -> Result<(), Box<dyn Error>>;
+}
+
+#[derive(Debug, Serialize)]
+struct FilesystemInMemory {}
+
+#[derive(Debug, Serialize, Clone, PartialEq)]
+struct NativeProvider<T: FileSystem + Debug> {
+    /// Namespace of the client
+    namespace:             String,
+    /// path where configuration relies
+    config_path:           String,
+    // variable that shows if debug is activated
+    debug:                 bool,
+    /// the timeout for the client to exit
+    timeout:               u32,
+    /// command sent to client
+    command:               String,
+    /// temporary directory
```

Comment:
Just my 2 cents but we should stay uniform in the doc comment, starting with uppercase + I don't think it's need to be doc domment given fields are private

Reply:
Fields on the struct are private, so the doc comment /// won't be available to end user and will be hidden in the doc, we can replace them with simple comments // and add uppercase on each beginning of the comment to keep it uniform with others comments